src/pykeen/triples/triples_factory.py

```diff
@@ -593,6 +659,121 @@ def split(
+            inference/evaluation graph.
+        :param evaluation_triples_ratios:
+            The split ratio for the inference graph split.
+        :param random_state:
+            The random state used to shuffle and split the triples.
+
+        :return:
+            A (transductive) training triples factory, the inductive inference triples factory,
+            as well as the evaluation triples factories.
+        """
+        training, inference, *evaluation = split_fully_inductive(
+            mapped_triples=self.mapped_triples,
+            entity_split_train_ratio=entity_split_train_ratio,
+            evaluation_triples_ratios=evaluation_triples_ratios,
+            random_state=random_state,
+        )
+        # separately condense the entity-to-id mappings for each of the graphs (training vs. inference)
+        # TODO: what about relations? We might not be allowed to do that!
+        training_tf = self.clone_and_exchange_triples(mapped_triples=training).condense()
+        inference_tf = self.clone_and_exchange_triples(mapped_triples=inference).condense()
+        # do not explicitly create inverse triples for testing; this is handled by the evaluation code
+        evaluation_tfs = [
+            inference_tf.clone_and_exchange_triples(mapped_triples=mapped_triples, create_inverse_triples=False)
+            for mapped_triples in evaluation
+        ]
+        # Make new triples factories for each group
+        return [training_tf, inference_tf] + evaluation_tfs
+
+    def merge(self, *others: Self) -> Self:
+        """Merge the triples factory with others.
+
+        The other triples factories have to be compatible.
+
+        :param others:
+            The other factories.
+
+        :return:
+            A new factory with the combined triples.
+
+        :raises ValueError:
+            If any of the other factories has incompatible settings
+            (number of entities or relations, or creation of inverse triples.)
+        """
+        if not others:
+            return self
+        mapped_triples = [self.mapped_triples]
+        for i, other in enumerate(others):
+            if other.num_entities != self.num_entities:
+                raise ValueError(
+                    f"Number of entities does not match for others[{i}]: {self.num_entities=} vs. {other.num_entities=}"
+                )
+            if other.num_relations != self.num_relations:
+                raise ValueError(
+                    f"Number of relations does not match for others[{i}]: "
+                    f"{self.num_relations=} vs. {other.num_relations=}"
+                )
+            if other.create_inverse_triples != self.create_inverse_triples:
+                raise ValueError(
+                    f"Creation of inverse triples does not match for others[{i}]: "
+                    f"{self.create_inverse_triples=} vs. {other.create_inverse_triples=}"
+                )
```

Comment:
is it worth checking if the entity mapping is available that the entities are actually the same? same for relations?

Reply:
https://github.com/pykeen/pykeen/pull/1416/commits/29431f8300472a0667982751a745c25e19e19387